docs/GHAFileExplained.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: page
+title: GitHub Actions CI workflow file explained
+---
+
+Below is the
+[gha.dist.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/master/gha.dist.yml)
+file with comments added to explain what each section is doing. For additional
+information please refer to [workflow syntax reference](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions).
+
+If you are familiar with Travis, this should be strightforward to understand
+the syntax, you may find this [migration
+manual](https://docs.github.com/en/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions)
+useful.
+
+```yaml
+# Title of the workflow
+name: Moodle Plugin CI
+
+# Run this workflow every time a new commit pushed to your repository or PR
+# created.
+on: [push, pull_request]
+
+jobs:
+  # Set the job key. The key is displayed as the job name
+  # when a job name is not provided
+  test:
+    # Virtual environment to use.
+    runs-on: ubuntu-18.04
```

Comment:
Do we want this or `ubuntu-latest` ?

Reply:
Hi @mudrd8mz,
not sure why but, maybe, something about code coverage should be there somewhere since it relates with Ubuntu flavour (`16.04`): more in https://github.com/moodlehq/moodle-plugin-ci/issues/62 and https://github.com/moodlehq/moodle-plugin-ci/pull/69.

HTH,
Matteo

Reply:
I think #62 can now be closed (#69 fixed it) given we create another issue for May, when Moodle 3.5 support ends, in order to bump to 18.04 then in those self.tests.

About what to use in the official example... I’m happy both with 16 and 18, maybe the later is better default with a note for people wanting to test coverage against php70.

but I think it’s better to, in any case, to use fixed/sticky version. In the past we have seen sudden problems when latest becomes updated with surprising changes.

just IMO

Reply:
> have seen sudden problems when latest becomes updated with surprising changes

Ah right, that's a valid argument. +1 thanks

Reply:
I made #62 a bit more clear to understand (also given recent discovery of actual logic there).